Sources/OAPoiUiFilterSettingsItem.mm

```diff
@@ -127,22 +128,38 @@ - (void) readItemsFromJson:(id)json error:(NSError * _Nullable __autoreleasing *
     }
 }
 
-- (void) writeItemsToJson:(id)json error:(NSError * _Nullable __autoreleasing *)error
+- (void) writeItemsToJson:(id)json
 {
     NSMutableArray *jsonArray = [NSMutableArray array];
     if (self.items.count > 0)
     {
         for (OAPOIUIFilter *filter in self.items)
         {
             NSMutableDictionary *jsonObject = [NSMutableDictionary dictionary];
-            jsonObject[@"name"] = filter.name;
-            jsonObject[@"filterId"] = filter.filterId;
-            jsonObject[@"acceptedTypes"] = [filter getAcceptedTypes];
+            jsonObject[kNAME_KEY] = filter.name;
+            jsonObject[kFILTER_ID_KEY] = filter.filterId;
+
+            NSMapTable<OAPOICategory *, NSMutableSet<NSString *> *> * acceptedTypes = [filter getAcceptedTypes];
```

Comment:
Seems like creating an NSDictionary with names and associated values and then converting it to JSON would be easier to read and less error-prone. Objective C has a pretty convenient way of converting collections to JSON string.